### PR TITLE
Change Jaeger CR to use es-self-provisioned approach - depends on es-…

### PR DESCRIPTION
**Pull request by objectiser**
…operator being deployed

Signed-off-by: Gary Brown <gary@brownuk.com>

### CODE REVIEW

**Comment by objectiser** [comment]
For completeness adding a draft PR that shows how the Jaeger CR should be changed to use the elasticsearch operator - however does not currently describe how to install this operator. Can be installed from operatorhub.

**Review by rcernich — APPROVED** [review]
lgtm

**Comment by objectiser** [comment]
@rcernich I've updated the README to include instructions for installing es-operator 4.1. Would you be able to test the changes against these instructions using the updated production-elasticsearch template?

Alternatively, are there instructions for how to best test the istio-operator from local changes to the helm chart? I've tested the jaeger operator against the es-operator and seems to work fine.

File: README.md [hunk 1]
```diff
@@ -4,20 +4,35 @@ This project is an operator (controller) that can be used to manage the installa
...
+Jaeger Operator, Elasticsearch Operator and Kiali Operator have been installed.
+
+### Installing the Elasticsearch Operator
+
+If available, the Elasticsearch operator (version 4.1) should be installed from the OperatorHub.
+
+Alternatively, to install the Elasticsearch operator manually, execute the following commands:
+
+```
+kubectl create ns openshift-logging # create the project for the elasticsearch operator
```
Comment (jkandasa) [inline README.md:18]: @objectiser should be OpenShift command, However, unable to create new namespace starts with "openshift-"
Seems we have some restriction. https://bugzilla.redhat.com/show_bug.cgi?id=1428934
```
$ oc new-project openshift-logging
Error from server (Forbidden): project.project.openshift.io "openshift-logging" is forbidden: cannot request a project starting with "openshift-"
```

Version: OCP 4.1.3

Comment (objectiser) [reply README.md:18]: This is the reason for using the kubectl command instead. This is only relevant for dev/test scenarios, as normally the elasticsearch operator would be installed via operatorhub - so should be ok. I'll update the PR to add a comment.

File: README.md [hunk 2]
```diff
@@ -4,20 +4,35 @@ This project is an operator (controller) that can be used to manage the installa
...
+oc create -f https://raw.githubusercontent.com/openshift/elasticsearch-operator/release-4.1/manifests/01-service-account.yaml -n openshift-logging
+oc create -f https://raw.githubusercontent.com/openshift/elasticsearch-operator/release-4.1/manifests/02-role.yaml
+oc create -f https://raw.githubusercontent.com/openshift/elasticsearch-operator/release-4.1/manifests/03-role-bindings.yaml
+oc create -f https://raw.githubusercontent.com/openshift/elasticsearch-operator/release-4.1/manifests/04-crd.yaml -n openshift-logging
+sed 's/latest/4.1/g' https://raw.githubusercontent.com/openshift/elasticsearch-operator/release-4.1/manifests/05-deployment.yaml | oc create -n openshift-logging -f -
```
Comment (jkandasa) [inline README.md]: @objectiser There is a minor change needs to update es operator version on the fly,
```
curl https://raw.githubusercontent.com/openshift/elasticsearch-operator/release-4.1/manifests/05-deployment.yaml | sed 's/latest/4.1/g' | oc create -n openshift-logging -f -
```

Comment (objectiser) [reply README.md]: Updated

**Comment by objectiser** [comment]
@rcernich If you could take another look - I've removed the old ES templates and added instructions to the readme, if you want to give them a try.

**Comment by objectiser** [comment]
@rcernich Could this be merged now?

**Comment by brian-avery** [comment]
Can you please add this to the docs too?

**Comment by objectiser** [comment]
@brian-avery Not sure what you mean? README has been updated - is there somewhere else?

**Comment by brian-avery** [comment]
Hi @objectiser These are the add (https://github.com/Maistra/maistra.github.io/blob/maistra-1.0/topics/docs/snippets/install_jaeger.adoc) and removal (https://github.com/Maistra/maistra.github.io/blob/maistra-1.0/topics/docs/snippets/remove_jaeger.adoc) docs for Jaeger.

**Comment by objectiser** [comment]
@brian-avery ok thanks - I'll do a PR now.

**Comment by brian-avery** [comment]
Thanks!

**Comment by objectiser** [comment]
https://github.com/Maistra/maistra.github.io/pull/65

**Pull request merged**